contracts/route/route.go

```diff
@@ -23,6 +24,10 @@ type Route interface {
 	RunTLSWithCert(host, certFile, keyFile string) error
 	// ServeHTTP serves HTTP requests.
 	ServeHTTP(writer http.ResponseWriter, request *http.Request)
+	// Listen starts the HTTP server and listens for incoming connections on the specified listener.
+	Listen(l net.Listener) error
+	// ListenTLS starts the HTTPS server with the provided TLS configuration and listens on the specified listener.
+	ListenTLS(l net.Listener, certFile, keyFile string) error
```

Comment:
This method is required because the std library uses an unexported method.

![image](https://github.com/user-attachments/assets/39c2f7c2-2068-4a9b-a8e4-9babda683972)